src/Microsoft.IdentityModel.Protocol.Extensions/Configuration/ConfigurationManager.cs

```diff
@@ -89,12 +89,18 @@ public ConfigurationManager(string metadataAddress, HttpClient httpClient)
         /// <param name="metadataAddress">the address to obtain configuration.</param>
         /// <param name="docRetriever">the <see cref="IDocumentRetriever"/> that reaches out to obtain the configuration.</param>
         public ConfigurationManager(string metadataAddress, IDocumentRetriever docRetriever)
+            : this(metadataAddress, null, docRetriever)
         {
-            if (!typeof(T).Equals(typeof(WsFederationConfiguration)) && (!typeof(T).Equals(typeof(OpenIdConnectConfiguration))))
-            {
-                throw new NotImplementedException(typeof(T).FullName);
-            }
+        }
 
+        /// <summary>
+        /// Instantiaties a new <see cref="ConfigurationManager{T}"/> that manages automatic and controls refreshing on configuration data.
+        /// </summary>
+        /// <param name="metadataAddress">the address to obtain configuration.</param>
+        /// <param name="configRetriever">the <see cref="IConfigurationRetriever{T}"/></param>
+        /// <param name="docRetriever">the <see cref="IDocumentRetriever"/> that reaches out to obtain the configuration.</param>
+        public ConfigurationManager(string metadataAddress, IConfigurationRetriever<T> configRetriever, IDocumentRetriever docRetriever)
```

Comment:
I don't like the design of having a public ctor that takes a parameter, and the ctor behaves differently if the parameter is null versus not null. I recommend enforcing that the configRetriever parameter isn't null in the ctor where it's a parameter. In the new ctor that doesn't have a configRetriever parameter, call the 3-param ctor as you are, but like this:
 
`public ConfigurationManager(string metadataAddress, IDocumentRetriever docRetriever)
: this(metadataAddress, GetConfigurationRetriever(), docRetriever)
          {
          }`

Tagging @brentschmaltz for his feedback.